### PR TITLE
Migrate from TSLint to ESLint

### LINKED ISSUE #3581: Migrate to eslint
Since TSLint is deprecated in 2019, typescript-eslint should be used as a successor. And I could help with migration. I did it in several of my projects, so it would be not so hard.
Pretty sure that config based on [@typescript-eslint/recommended](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#recommended-configs) plugin will fit perfectly
What do you think?

### PR DESCRIPTION
**Pull request by standy**
Related: #3581 

Migrated from TSLint to ESLint. 
Used `@typescript-eslint/recommended` + `standard` configs with `prettier` integration

### CODE REVIEW

**Comment by dotansimha** [comment]
This is awesome!!! Thank you so much @standy ! 

@kamilkisiela @ardatan can you please take a look? 

@standy some CI tasks are failing for List task, not sure why, can you please take a look?

**Comment by standy** [comment]
Still working on it. There are some type errors during build

File: packages/graphql-codegen-cli/src/utils/listr-renderer.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export class Renderer {
   ) {
     this.updateRenderer.end(err);
 
-    if (typeof err === undefined) {
+    if (typeof err === 'undefined') {
```
Comment (standy) [inline packages/graphql-codegen-cli/src/utils/listr-renderer.ts]: Not sure if this is correct, `err` has an Error type, so cant be `undefined`.
Very suspicious

File: packages/graphql-codegen-cli/tests/codegen.spec.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ describe('Codegen Executor', () => {
       try {
         await executeCodegen({
           schema: join(__dirname, './test-files/schema-dir/schema-object.js'),
-          require: join(__dirname + './missing.js'),
+          require: join(__dirname, './missing.js'),
```
Comment (standy) [inline packages/graphql-codegen-cli/tests/codegen.spec.ts]: Pretty sure that fixed something, good job eslint

File: packages/plugins/typescript/type-graphql/src/index.ts [hunk 3]
```diff
@@ -19,9 +19,9 @@ export const plugin: PluginFunction<TypeGraphQLPluginConfig> = (schema: GraphQLS
   const introspectionDefinitions = includeIntrospectionDefinitions(schema, documents, config);
   const scalars = visitor.scalarsDefinition;
 
-  let definitions = visitorResult.definitions;
+  const definitions = visitorResult.definitions;
   // Sort output by interfaces first, classes last to prevent TypeScript errors
-  definitions.sort((definition1, definition2) => (isDefinitionInterface(definition1) ? -1 : 1));
+  definitions.sort((definition1, definition2) => +isDefinitionInterface(definition2) - +isDefinitionInterface(definition1));
```
Comment (standy) [inline packages/plugins/typescript/type-graphql/src/index.ts]: Compare function should return the same value for the same elements. If compare result for `(definition1, definition2)` is `1`, then result for `(definition2, definition1)` should be `-1`

**Comment by standy** [comment]
> @standy some CI tasks are failing for List task, not sure why, can you please take a look?

@dotansimha  Done 

Ready to review @kamilkisiela @ardatan

**Comment by ardatan** [comment]
@standy This is awesome; 👍🏻 LGTM

**Pull request merged**